orderbook/src/main.rs

```diff
@@ -127,6 +127,21 @@ struct Arguments {
         parse(try_from_str = shared::arguments::duration_from_seconds),
     )]
     solvable_orders_max_update_age: Duration,
+
+    /// Used to specify additional fee subsidy factor based on app_ids contained in orders.
+    /// Should take the form of a json string as shown in the following example:
+    /// '{"0x0000000000000000000000000000000000000000000000000000000000000000":0.5,"$PROJECT_APP_ID":0.7}'
+    ///
+    /// Furthermore, a value of
+    /// - 1 means no subsidy and is the default for all app_data not contained in this list.
+    /// - 0.5 means that this project pays only 50% of the estimated fees.
+    #[structopt(
+        long,
+        env,
+        default_value = "{}",
+        parse(try_from_str = serde_json::from_str),
+    )]
+    partner_additional_fee_factors: HashMap<H256, f64>, // '{"$BALANCER_APP_ID":0.5,"$METAMASK_APP_ID":0.7}'
```

Comment:
Is using json in a command line argument a good way to implement this? It is a small amount of code but it feels weird to me that you would pass json. More natural I would find something like
--partner-additional-fee-factors "0x0000000:0.5,0x0000:0.3".
That is more code to implement though. You would need a custom parse function that goes from string (the raw argument) to the hashmap.
Not sure how others feel about this.

Reply:
I had initially thought of going with a list of tuples (as you suggested), but thought it was cute/nice to go directly to the hashmap that we wanted. Will wait on others as well.

Reply:
We used JSON in v1 and it was OK. I agree that its not ideal, but I do slightly prefer it to writing custom deserialization code. No strong opinion either.